daggerverse/gcp-secrets/main.go

```diff
@@ -0,0 +1,146 @@
+// A generated module for GcpSecrets functions
+
+package main
+
+import (
+	"context"
+	"dagger/gcp-secrets/internal/dagger"
+	"errors"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"google.golang.org/api/option"
+)
+
+type GcpSecrets struct{}
+
+// Get a single secret from the Google Secret Manager with latest version
+// example: dagger call get-secret --project-id=project-id --secret-name=secret-name --version=latest --credential=path-to-Servcie-account-file
+func (m *GcpSecrets) GetSecretWithVersion(projectID string, secretName string, version string, credential *dagger.File) (string, error) {
+
+	// Create the client.
+	ctx := context.Background()
+
+	json, err := credential.Contents(ctx)
+	if err != nil {
+		return "", err
+	}
+	b := []byte(json)
+	client, err := secretmanager.NewClient(ctx, option.WithCredentialsJSON(b))
+	if err != nil {
+		return "", err
+	}
+	defer client.Close()
+
+	// Create the request to create the secret.
+	secretName = "projects/" + projectID + "/secrets/" + secretName + "/versions/" + version
+
+	// Build the request.
+	accessRequest := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: secretName,
+	}
+
+	// Call the API.
+	result, err := client.AccessSecretVersion(ctx, accessRequest)
+	if err != nil {
+		return "", err
+	}
+
+	// Print the secret payload.
+	return string(result.Payload.Data), nil
```

Comment:
This needs to return a `dagger.Secret` to keep the value secure

https://docs.dagger.io/manuals/developer/secrets/